bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -17,6 +16,8 @@ import {sum} from "core/util/array"
 import {isNumber} from "core/util/types"
 import {GraphicsBoxes, TextBox} from "core/graphics"
 import {Factor, FactorRange} from "models/ranges/factor_range"
+import {color2css} from "core/util/color"
+import {MathText} from "models/tools/math_text"
```

Comment:
Definitively `MathText` doesn't belong `models/tools`. To be honest, I'm not sure where it should be categorized currently, because it doesn't fit neither annotations nor generic renderers. This is one reason why `TextBox` is currently not a model but an internal class.

Reply:
I think it fits better as a GraphicsBox subclass, but would it be right to use it as the Axis.label type? How do I get its instance type if its not a Model that is created when reading the JSON generated from python side?

Reply:
I agree it doesn't belong there, it looks a lot like TextBox, it needs a lot of its sizing calculations, but the key difference is that it needs to keep the reference to the SVG between renders, which TextBox does not. Would it be right to use something like a MathTextBox? How would I instantiate the class like we do with the models?

Reply:
@bryevdv we talked about similar questions in #11191 , what's your take on this one?

Reply:
One option would be to create something like a corresponding "Plain Text" model and use that wherever  plain text is currently provided as a bare string. We would need to afford conveniences ("accepts", etc) so that users could still supply bare strings everywhere they do now. Then both these models could go in `models/text` or similar. This gets back to my comment here https://github.com/bokeh/bokeh/pull/11128#discussion_r644197163 about abstracting text rendering so that things like axes, labels, titles, etc don't have to do explicit switching internally